tensorflow/contrib/seq2seq/python/ops/seq2seq.py

```diff
@@ -0,0 +1,111 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Seq2seq layer operations for use in neural networks.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.ops import tensor_array_ops
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import variable_scope as vs
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import control_flow_ops
+from tensorflow.python.ops import rnn
+
+from tensorflow.contrib import layers
+
+__all__ = ["rnn_decoder"]
+
+def rnn_decoder(cell, decoder_fn, inputs=None,
+                    sequence_lengths=None,
+                    parallel_iterations=None,
+                    swap_memory=False, time_major=False, scope=None):
+  if inputs is not None:
+    # Testing input dimensions
+    if len(inputs.get_shape()) is not 3:
+      raise ValueError("Inputs must have three dimensions")
+    if inputs.get_shape()[2] is None:
+      raise ValueError("Inputs must not be `None` in the feature (3'rd) "
+                       "dimension")
+    # Setup of RNN (dimensions, sizes, length, initial state, dtype)
+    if not time_major:
+      # [batch, seq, features] -> [seq, batch, features]
+      inputs = array_ops.transpose(inputs, perm=[1, 0, 2])
+
+    dtype = inputs.dtype
+    # Get data input information
+    input_depth = int(inputs.get_shape()[2])
+    max_time = array_ops.shape(inputs)[0]
+    # Setup decoder inputs as TensorArray
+    inputs_ta = tensor_array_ops.TensorArray(dtype, size=0, dynamic_size=True)
+    inputs_ta = inputs_ta.unpack(inputs)
+
+  def loop_fn(time, cell_output, cell_state, loop_state):
+    if cell_state == None: # first call, before while loop (in raw_rnn)
+      assert cell_output is None
+      assert loop_state is None 
+      context_state = None
+    else: # subsequent calls, inside while loop, after cell excution 
+      if isinstance(loop_state, tuple):
+        (done, context_state) = loop_state
+      else:
+        done = loop_state
+        context_state = None
+
+    # call decoder function
+    if inputs is not None: # training
+      # get next_cell_input
+      if cell_state == None:
+        next_cell_input = inputs_ta.read(0)
+      else:
+        batch_size = array_ops.shape(done)[0]
+        next_cell_input = control_flow_ops.cond(
+            tf.equal(time, max_time),
```

Comment:
you're inside the "import tf" so you can't use tf here.  rather you have to import the specific file which contains equal; in this case math_ops:

from tensorflow.python.ops import math_ops
and here use math_ops.equal(...)

same for every place you use tf...

basically remove the "import tensorflow as tf" at the top and fix any breakages like this.